### PR TITLE
Complete development of dropdown

### PR DESCRIPTION
**Pull request by jordanjones243**
# Alaska Airlines Pull Request

Resolves: #20, #25

## Before Submitting this pull request:
- Link all tickets in this repository related to this PR in the `Development` section
_note: all pull requests require at least one linked ticket_
- If this PR is `Ready For Review`, all ticket's linked under `Development` must have their status changed to `Ready For Review` as well

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license and I have performed a self-review of my own update.**

## Summary by Sourcery

Refactor the AuroDropdown component by removing deprecated events and unused code, and enhance the AuroDropdownBib component with new styling properties. Update the documentation to reflect these changes and improve clarity.

Enhancements:
- Refactor the AuroDropdown component to remove deprecated events and unused code, improving maintainability.
- Add new properties to AuroDropdownBib for styling options like common, rounded, and inset, enhancing customization.

Documentation:
- Update README.md to reflect new documentation paths and remove outdated content.

## Summary by Sourcery

Complete the development of the dropdown component by refactoring the AuroDropdown to remove deprecated events and unused code, and enhancing the AuroDropdownBib with new styling properties. Update the documentation to reflect these changes.

New Features:
- Introduce new styling properties for AuroDropdownBib, including common, rounded, and inset, to enhance customization options.

Enhancements:
- Refactor AuroDropdown component by removing deprecated events and unused code to improve maintainability.

Documentation:
- Update README.md to reflect new documentation paths and remove outdated content.

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# auro-dropdownbib
```
Comment (sun-mota) [inline docs/api.md:1]: this needs to be under `components/dropdown/docs`, I assume?

File: components/dropdown/demo/api.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 /* eslint-disable jsdoc/require-jsdoc, no-magic-numbers */
 import { showExample } from './../apiExamples/programmaticallyShow.js';
 import { hideExample } from './../apiExamples/programmaticallyHide.js';
+import './index.js';
```
Comment (sun-mota) [inline components/dropdown/demo/api.js:4]: `./index.js` also has `initExamples` function which does exactly the same. It might be good to utilize that.

**Review by sun-mota — CHANGES_REQUESTED** [review]

**Comment by sun-mota** [comment]
a little thing. maybe add some explanation how to restrict the height. re: https://github.com/AlaskaAirlines/auro-form/issues/25

File: components/dropdown/src/auro-dropdown.js [hunk 3]
```diff
@@ -22,16 +21,11 @@ import styleCss from "./styles/style-css.js";
 import colorCss from "./styles/color-css.js";
```
Comment (sun-mota) [inline components/dropdown/src/auro-dropdown.js:21]: iconVersion should be from `root/src/iconVerion.js` and `components/dropdown/src/iconVersion.js` should be removed

File: components/dropdown/demo/index.min.js [hunk 4]
```diff
@@ -1,8 +1,2719 @@
...
+
+// Copyright (c) 2020 Alaska Airlines. All right reserved. Licensed under the Apache-2.0 license
+// See LICENSE in the project root for license information.
+
+
+// See https://git.io/JJ6SJ for "How to document your components using JSDoc"
+/**
+ * auro-icon provides users a way to use the Auro Icons by simply passing in the category and name.
+ *
+ * @attr {Boolean} customSize - Allows for custom size use.
```
Comment (sun-mota) [inline components/dropdown/demo/index.min.js:2107]: seems like the bundler is not utilizing the latest auro-icon as this file shows `customize`

Comment (jordanjones243) [reply components/dropdown/demo/index.min.js:2107]: Something is wrong with my system, I will need your assistance with correcting this bug.

File: docs/api.md (hunk 1 above)
Comment (jordanjones243) [reply docs/api.md]: This will be addressed here:

-https://github.com/AlaskaAirlines/auro-form/issues/30

**Pull request merged**